certbot/certbot/_internal/client.py

```diff
@@ -49,8 +51,12 @@ def acme_from_config_key(config: configuration.NamespaceConfig, key: jose.JWK,
                          regr: Optional[messages.RegistrationResource] = None
                          ) -> acme_client.ClientV2:
     """Wrangle ACME client construction"""
-    # TODO: Allow for other alg types besides RS256
-    net = acme_client.ClientNetwork(key, account=regr, verify_ssl=(not config.no_verify_ssl),
+    if key.typ in ('EC', 'OKP'):
+        alg = ES256
```

Comment:
I don't think this quite works. If you generate a certificate with a curve other than the default curve, we blow up later with an error like:
```
acme.messages.Error: urn:ietf:params:acme:error:badSignatureAlgorithm :: The JWS was signed with an algorithm the server does not support :: JWS signature header algorithm "ES256" does not match expected algorithm "ES384" for JWK
```
This is because https://datatracker.ietf.org/doc/html/rfc7518#section-3.4 says that the longer SHA algorithms must be used with the larger curves.

Reply:
Yes, that's a very valid point. I am experimenting a bit with a possible solution, and will push something once I get the tests to pass.